### ISSUE TITLE
Adding new button widgets results in too-wide buttons

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#6259]
ref: https://github.com/bokeh/bokeh/issues/5856

Hopefully this is something simple, like a re-compute not being triggered when it should 

ping @mattpap 

code:
```
import numpy as np

from bokeh.io import curdoc
from bokeh.layouts import row
from bokeh.models import ColumnDataSource
from bokeh.plotting import figure

# Set up data
x = np.linspace(0, 4*np.pi, 200)
y = np.sin(x)
source = ColumnDataSource(data=dict(x=x, y=y))

# Set up plot
plot = figure()

plot.line('x', 'y', source=source)

# Problem Code:
from bokeh.models.widgets import Button
from bokeh.models.layouts import VBox

button = Button(label="Add")
vbox = VBox(children=[button])

def create_button():
    button = Button(label=str(len(vbox.children)))
    vbox.children.append(button)

button.on_click(create_button)

curdoc().add_root(row(vbox, plot))
curdoc().title = "Sliders"
```

Results in:

<img width="980" alt="screen shot 2017-05-08 at 17 25 39" src="https://cloud.githubusercontent.com/assets/1078448/25828406/9907594c-3415-11e7-9f56-86a697fbbec3.png">

**Comment by bryevdv** [comment]
Just confirming that this is not fixed by any work up to `0.12.6dev7`

**Comment by mattpap** [comment]
This isn't a problem with layout per se, but with its API. Widget have to be wrapped in `WidgetBox` to make them appear properly (I'm not sure why this is needed). The following code works:
```
def create_button():
    from bokeh.models.layouts import WidgetBox
    button = Button(label=str(len(vbox.children)))
    vbox.children.append(WidgetBox(children=[button]))
```
The problem is that some API functions that work with layout do this automatically, but if you append to children like this, you have to do wrapping manually.

**Comment by bryevdv** [comment]
Oh, another probably slightly better is to put all the buttons in one widget box

**Comment by bryevdv** [comment]
fixed in #8085:

<img width="947" alt="screen shot 2019-01-30 at 21 20 34" src="https://user-images.githubusercontent.com/1078448/52032207-e844cb00-24d4-11e9-8cd0-624398545040.png">
